### PR TITLE
Ensure ACLs never raise errors from API responses

### PR DESCRIPTION
**Pull request by haakonvt**
## Description
Ensures all error handling still runs for users instantiating their capabilities, but let's unknown through when they come straight from API responses.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: tests/tests_unit/test_data_classes/test_capabilities.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def unknown_cap_with_extra_key():
 class TestCapabilities:
```
Comment (erlendvollset) [inline tests/tests_unit/test_data_classes/test_capabilities.py:135]: Let's add some test cases around this? unknown scopes/actions/acls

File: cognite/client/data_classes/capabilities.py [hunk 2]
```diff
@@ -137,8 +148,9 @@ def load(cls, resource: dict | str) -> Self:
             return cast(
                 Self,
                 capability_cls(
-                    actions=[capability_cls.Action(action) for action in resource[name]["actions"]],
+                    actions=[capability_cls.Action.load(act, allow_unknown) for act in resource[name]["actions"]],
```
Comment (doctrino) [inline cognite/client/data_classes/capabilities.py:151]: What about just monkey patching? Would you lose information with this solution? The action is lost.

Comment (haakonvt) [reply cognite/client/data_classes/capabilities.py:151]: Not sure I understood your suggestion. For the record, I just copied this one:

IMG_1567

Comment (haakonvt) [reply cognite/client/data_classes/capabilities.py:151]: But yeah, I get your point about losing the action. Hmmm

File: tests/tests_unit/test_data_classes/test_capabilities.py (hunk 1 above)
Comment (haakonvt) [reply tests/tests_unit/test_data_classes/test_capabilities.py:141]: Yeah, just waiting to see if you guys had a better idea for implementation 😅

File: cognite/client/data_classes/capabilities.py (hunk 2 above)
Comment (haakonvt) [reply cognite/client/data_classes/capabilities.py:159]: Switched to dynamic enums for when _unknowns_ are encountered to ensure we don't lose information. Too crazy?

File: tests/tests_unit/test_data_classes/test_capabilities.py (hunk 1 above)
Comment (haakonvt) [reply tests/tests_unit/test_data_classes/test_capabilities.py:142]: ...done!

**Review by doctrino — APPROVED** [review]

**Pull request merged**